services/server/server.go

```diff
@@ -460,6 +461,12 @@ func (s *Server) validateChangeSettingsRequest(ctx context.Context, req *serverp
 		return status.Error(codes.FailedPrecondition, "Data retention for queries is set via DATA_RETENTION environment variable.")
 	}
 
+	if req.DisableStt && req.EnableStt {
+		return status.Error(codes.InvalidArgument, "Enable STT and disable STT cannot be both true")
+	}
+	if req.EnableStt && !req.EnableTelemetry {
```

Comment:
let's check that `req.EnableStt` is not passed with `req.DisableTelemetry` and `req.DisableStt` is not passed with `req.EnableTelemetry`. Do not forget about `s.envSettings.DisableTelemetry`